api/endpoints/property.py

```diff
@@ -18,73 +35,129 @@ def get_properties():
     return properties_schema.jsonify(properties)
 
 
-@property_endpoint.route('/api/property/<int:prop_id>', methods=['GET'])
+@property_endpoint.route("/api/property/<int:prop_id>", methods=["GET"])
 def get_one_property(prop_id):
     """
     Returns a json object of one property matching the ID.
     """
     prop_data = Property.query.get(prop_id)
     if not prop_data:
-        return jsonify({"message": f"The property listing with ID {prop_id} was not found."})
+        return jsonify(
+            {"message": f"The property listing with ID {prop_id} was not found."}
+        )
     return property_schema.jsonify(prop_data)
 
 
-@property_endpoint.route('/api/property/add', methods=['POST'])
+@property_endpoint.route("/api/property/add", methods=["POST"])
+@jwt_required
 def add_property():
     """
     Creates a new property in the database with data from request body.
     """
-    data = request.get_json()
+    data = request.form
+    current_user = get_jwt_identity()
+    if "photos" not in data and request.files["photos"].filename == "":
+        return jsonify({"message": "You need to upload photos."})
     try:
+        image_files = request.files.getlist("photos")
+        folder_to_save_images = create_images_folder(current_user)
+        # List of image filenames to save to database
+        image_list = property_image_handler(image_files, folder_to_save_images)
+
+        # convert the python dictionary(image_files) to json string
+        image_list_to_json = json.dumps(image_list)
+        # Check if the request data matches the schema else raise ValidationError
         prop_data = property_schema.load(data)
         prop_to_add = Property(
-            name=prop_data['name'],
-            desc=prop_data['desc'],
-            price=prop_data['price'],
-            location=prop_data['location'],
-            photos=prop_data['photos'],
-            user_id=prop_data['user_id']
+            name=prop_data["name"],
+            desc=prop_data["desc"],
+            price=prop_data["price"],
+            location=prop_data["location"],
+            image_folder=folder_to_save_images,
+            photos=image_list_to_json,
+            user_id=prop_data["user_id"],
         )
         db.session.add(prop_to_add)
         db.session.commit()
```

Comment:
I think we can optimize this by moving these actions as methods under specific DB models
I think it will be cleaner calling the actions from there
what do you think?

Reply:
That's a good idea and it will be much cleaner